src/adapters/leasing-adapter.ts

```diff
@@ -447,22 +448,27 @@ const getApplicantByContactCodeAndListingId = async (
   }
 }
 
-// TODO: This function does not actually get the listing. Rename and describe function?
-const getListingByIdWithDetailedApplicants = async (
-  listingId: string
-): Promise<DetailedApplicant[] | undefined> => {
+const getDetailedApplicantsByListingId = async (
+  listingId: number
+): Promise<AdapterResult<DetailedApplicant[], 'unknown' | 'not-found'>> => {
   try {
     const response = await axios(
       `${tenantsLeasesServiceUrl}/listing/${listingId}/applicants/details`
     )
-    return response.data.content
+
+    return { ok: true, data: response.data.content }
   } catch (error) {
-    logger.error(
-      error,
-      'Error fetching listing with detailed applicant data:',
-      error
-    )
-    return undefined
+    logger.error(error, 'Error fetching detailed applicant data by listing id')
+
+    if (!(error instanceof AxiosError)) {
+      return { ok: false, err: 'unknown' }
+    }
+
+    if (error.response?.status === 404) {
+      return { ok: false, err: 'not-found' }
```

Comment:
We should get the error-codes from types and not have them as strings.

Reply:
What about add those error codes to this endpoint when we need them? I did actually re-write the endpoint but adding error codes here and in core feels a bit out of scope.